eureka/S5_lightcurve_fitting/lightcurve.py

```diff
@@ -185,28 +196,55 @@ def plot(self, meta, fits=True):
         None
         """
         # Make the figure
-        fig = plt.figure(int('54{}'.format(str(self.channel).zfill(len(str(self.nchannel))))), figsize=(8,6))
-        fig.clf()
-        # Draw the data
-        ax = fig.gca()
-        ax.errorbar(self.time, self.flux, self.unc, fmt='.', color=next(COLORS), zorder=0)
-        # Draw best-fit model
-        if fits and len(self.results) > 0:
-            for model in self.results:
-                model.plot(self.time, ax=ax, color=next(COLORS), zorder=np.inf)
-
-        # Format axes
-        ax.set_xlabel(str(self.time_units))
-        ax.set_ylabel('Flux')
-        ax.legend(loc='best')
-        fig.tight_layout()
-
-        fname = 'figs/fig54{}_all_fits.png'.format(str(self.channel).zfill(len(str(self.nchannel))))
-        fig.savefig(meta.outputdir+fname, bbox_inches='tight', dpi=300)
-        if meta.hide_plots:
-            plt.close()
+        if self.share:
```

Comment:
This if/else pair seems like it could easily be merged so that unshared fits are just a special case. As of right now there is so much duplicated code

Reply:
Done